pandas/core/indexes/base.py

```diff
@@ -3872,7 +3893,14 @@ def _get_indexer(
             else:
                 tgt_values = target._get_engine_target()
 
```

Comment:
should this check be for maskedness instead of EA-ness?

Reply:
Oh yeah good point. ``tgt_values`` are already cast to object in case of ea with non masked, but we should be explicit